File: backend-carrogerenciamento/src/main/java/com/felipe/carros/carrogerenciamento/entity/Marca.java
package com.felipe.carros.carrogerenciamento.entity;

import com.fasterxml.jackson.annotation.JsonIgnore;
import com.fasterxml.jackson.annotation.JsonManagedReference;
import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;
import org.springframework.web.service.annotation.GetExchange;

import java.util.List;

@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Entity
public class Marca {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private long id;
    private String nome;
    @OneToMany(mappedBy = "marca")
    @JsonIgnore
    private List<Carro> carros;
}

File: frontend-carrogerenciamento/src/app/models/marca.ts
import { Carro } from './carro';

export class Marca {
  id!: number;
  nome!: string;
  carros!: Carro[];

  constructor(id: number, nome: string, carros: Carro[]) {
    this.id = id;
    this.nome = nome;
    this.carros = carros;
  }
}
